test/unit/test_archive.py

```diff
@@ -581,6 +581,12 @@ def do_MKCOL(self):
                 self.send_response(403)
             self.end_headers()
 
+        def do_OPTIONS(self):
```

Comment:
That could go to a preparatory commit. Otherwise, the test adaptations, should be in the same commit that changes the code. The goal is that is must be possible to `git bisect` and run the tests at any stage...